code/modules/mob/living/carbon/human/appearance.dm

```diff
@@ -72,44 +54,59 @@
 
 /mob/living/carbon/human/proc/change_head_accessory(var/head_accessory_style)
 	var/obj/item/organ/external/head/H = get_organ("head")
-	if(!head_accessory_style)
-		return
-
-	if(H.ha_style == head_accessory_style)
-		return
-
-	if(!(head_accessory_style in head_accessory_styles_list))
+	if(!head_accessory_style || H.ha_style == head_accessory_style || !(head_accessory_style in head_accessory_styles_list))
 		return
 
 	H.ha_style = head_accessory_style
 
 	update_head_accessory()
 	return 1
 
-/mob/living/carbon/human/proc/change_markings(var/marking_style)
-	if(!marking_style)
+/mob/living/carbon/human/proc/change_markings(var/marking_style, var/location = "body")
+	var/list/marking_styles = params2list(m_styles)
+	if(!marking_style || marking_styles[location] == marking_style || !(marking_style in marking_styles_list))
 		return
 
-	if(src.m_style == marking_style)
+	var/datum/sprite_accessory/body_markings/marking = marking_styles_list[marking_style]
+	if(marking.name != "None" && marking.marking_location != location)
 		return
 
-	if(!(marking_style in marking_styles_list))
-		return
+	var/obj/item/organ/external/head/head_organ = get_organ("head")
+	if(location == "head")
+		if(head_organ.alt_head && head_organ.alt_head != "None")
+			var/datum/sprite_accessory/body_markings/head/H = marking_styles_list[marking_style]
+			if(marking.name != "None" && (!H.heads_allowed || !(head_organ.alt_head in H.heads_allowed)))
+				return
+		else
+			if(!head_organ.alt_head || head_organ.alt_head == "None")
+				head_organ.alt_head = "None"
+				var/datum/sprite_accessory/body_markings/head/H = marking_styles_list[marking_style]
+				if(H.heads_allowed )
+					return
+
+	if(location == "tail" && marking.name != "None")
+		var/datum/sprite_accessory/body_markings/tail/tail_marking = marking_styles_list[marking_style]
+		if(!body_accessory)
+			if(tail_marking.tails_allowed)
+				return
+		else
+			if(!tail_marking.tails_allowed || !(body_accessory.name in tail_marking.tails_allowed))
+				return
 
-	src.m_style = marking_style
+	marking_styles[location] = marking_style
+	m_styles = list2params(marking_styles)
 
-	update_markings()
+	if(location == "tail")
+		stop_tail_wagging()
+	else
+		update_markings()
 	return 1
 
 /mob/living/carbon/human/proc/change_body_accessory(var/body_accessory_style)
 	var/found
-	if(!body_accessory_style)
+	if(!body_accessory_style || (src.body_accessory && src.body_accessory.name == body_accessory_style))
```

Comment:
`src.` isn't needed, and shouldn't ever be, assuming coders don't name their proc arguments the same as their variables.